### DISCUSSION TITLE
Can't import a lot of my CRP Files

### DISCUSSION THREAD

**Discussion by ovahead** [mytourbook/mytourbook#924] [General]
As I wrote before on Sourceforge I can't import some of my CRP Files (CicloTour/HAC4). That file that Frederic used was okay. But I think he didn't saw the other file:
220322.crp is one of the files that won't import.

Sorry Frederic for wrote that in the discussion where the thread was for Fit Files. 

[220322.zip](https://github.com/mytourbook/mytourbook/files/9485537/220322.zip)

![MyTourbook-Tour-Protocol](https://user-images.githubusercontent.com/112829463/188334822-3c9c37f8-2b1b-44ab-8927-1d21e0ba92a1.jpg)

Even if I can import a file all values without distance and the total time are different. Starts with the moving time.

**Comment by wolfgang-ch** [comment]
The attached file do not look like a supported crp file

![image](https://user-images.githubusercontent.com/1283445/188362238-320dd592-3a30-404e-b33d-eb2dbf89d873.png)

![image](https://user-images.githubusercontent.com/1283445/188362860-94cca8c9-dfce-4103-be72-09d72d1c7bc6.png)

**Comment by ovahead** [comment]
Thanks and what is with that file that Frederic and I could import. That file look similar, is it also not in ASCII?

Copied and changed crp to txt:
[080522.txt](https://github.com/mytourbook/mytourbook/files/9489935/080522.txt)

BTW Both files imported this year from a HAC4 to CicloTour 4.42.

**Reply by FJBDev** [reply]
> Thanks and what is with that file that Frederic and I could import. That file look similar, is it also not in ASCII?
> 
 I was never able to import both files you put on sourceforge if that is what youre talking about, both files failed to import on 22.8.0

**Reply by wolfgang-ch** [reply]
> Copied and changed crp to txt

This does not change the content of the file which should look similar like this

![image](https://user-images.githubusercontent.com/1283445/188548572-4513eda2-4a74-4597-bb00-08eaa7c35955.png)

**Comment by ovahead** [comment]
I also tried an export from CicloTour as txt file. The saved file is indeed a txt file and not a CSV file (as described in Supported Formats). But whatever the extension is in this case no import from Mytourbook 
Tried a data transfer from CicloTour to Excel and then saved it as MS-Dos CSV. Still no import from Mytourbook

**Comment by ovahead** [comment]
Here is a CicloTour Text File generated with CicloTour 4.42 that nevertheless can't import Mytourbook:
[220322.txt](https://github.com/mytourbook/mytourbook/files/9491920/220322.txt)

![Mytourbook-Ciclotour-Txt-Import](https://user-images.githubusercontent.com/112829463/188510658-f040c0f2-5f99-4ab1-8c80-4110a0ec09f3.jpg)

**Comment by FJBDev** [comment]
My understanding is that crp filed have to start with [this](https://github.com/mytourbook/mytourbook/blob/main/bundles/net.tourbook.device/src/net/tourbook/device/crp/CRPDataReader.java#L445) , but this file format is new to me

**Comment by FJBDev** [comment]
@ovahead can you give more details of your issue?
Is it a new problem? Was it working before? If yes, when did it stop working (version of MT) ?

Thanks
Frederic

**Comment by wolfgang-ch** [comment]
Link to the original discussion

https://sourceforge.net/p/mytourbook/discussion/622811/thread/d52269ce04/?limit=25#cef1

**Comment by ovahead** [comment]
Frederic, it's all new to me because my HAC4 is broken and now I have an Edge 830. Even MyTourbook is new to me. So I'll try to import all my CRP Files (2003-2022) in Mytourbook.
I talked with Ralph from CicloTour/HRMProfil/TrainingLab and he said that later CRP files (since 2006 or 2008) were compressed with ZLib.

So the problem is some of this Zlib compressed CRP Files I can import in M.T.B. and some I are not.

edit:
Curios the 080522.CRP is imported but I can't import it again... I'll try with an other file to give a prove. 

Ralph said also that the MyTourbook Ciclosport Plugin could be easily changed so that all crp files could be decompressed and imported. He gave me this link:
https://docs.oracle.com/javase/8/docs/api/java/util/zip/Deflater.html ;)

Cheers Stefan (ovahead)

**Reply by FJBDev** [reply]
If some crp files are compressed, then shouldnt their extension be in zip or something similar?

**Reply by wolfgang-ch** [reply]
> If some crp files are compressed, then shouldnt their extension be in zip or something similar?

No, I also have some of these compressed .crp files and both files, the ASCII and compressed, have the same .crp extension. I didn't knew that the compressed file is simply compressed with Zlib, otherwise I would have implemented it. But this should be now an easy task for you :-)

**Comment by ovahead** [comment]
Okay a prove that the file was onced before imported (and of course other Zlib compressed files, Tour 15,16,17)  
![Tour-080522-in-Ciclotour-](https://user-images.githubusercontent.com/112829463/188647933-46a42f46-5c23-42f0-b602-d613b8c58000.jpg)
![Tour-080522-in-Mytourbook](https://user-images.githubusercontent.com/112829463/188650000-0cf39898-c08c-49cb-b011-3cfa7e8a8c2a.jpg)

**Reply by FJBDev** [reply]
Can you attach that crp file so i can see what a "valid" file looks like?
Thanks

**Comment by ovahead** [comment]
As I wrote already yesterday a second option could be to export the files to text files and import them in MyTourbook. But unfortunately it's also not working.
https://github.com/mytourbook/mytourbook/discussions/924#discussioncomment-3563991

**Reply by FJBDev** [reply]
Is there a way to export in different format such as tcx, fit...etc?

**Comment by WRPSoft** [comment]
Hi to all,

Stefan asked me for some infos yesterday. I'm the developer of HRMProfil/CicloTour/TrainingLab Pro.
I don't have access to a Java compiler right now and just typed the few lines into jdoodle (https://www.jdoodle.com/online-java-compiler/) online during my breakfast break. So please ignore the ugly code. I'm not a Java professional developer!

Basically, all you have to do is call the ZLibCompression.decompress method for decompressing the compressed crp files.

Maybe it helps, if someone wants to tackle this special import type again.

This java code should do the trick. 

Regards Ralph

--- elekt. Beisskannte ---

import java.util.Scanner;
import java.util.zip.*;
import java.io.*;

public class MyClass
{
  public static void main (String args[])
  {

    if (args.length < 1) {
      return;
    }
    
    String fname = args[0];
    System.out.println (fname);

    File mfile_in = new File (fname);

    int i = mfile_in.getName ().lastIndexOf ('.');
    String newfname = mfile_in.getName ().substring (0, i);
    // not allowed within online compiler, read only file
    // File mfile_out = new File (mfile_in.getParent(), newfname + "_.crp");    
    File mfile_out = new File ("/myfiles/output.crp");	
    try
    {
      ZLibCompression.decompress (mfile_in, mfile_out);
    }
    catch (IOException e)
    {
      e.printStackTrace ();
    }
  }


  class ZLibCompression
  {

    public static void decompress (File compressed, File raw)
      throws IOException
    {
      try (InputStream inputStream =
	   new InflaterInputStream (new FileInputStream (compressed));
	   OutputStream outputStream = new FileOutputStream (raw))
      {
	copy (inputStream, outputStream);
      }
    }

    public static String decompress (File compressed) throws IOException
    {
      try (InputStream inputStream =
	   new InflaterInputStream (new FileInputStream (compressed)))
      {
	return toString (inputStream);
      }
    }

    private static String toString (InputStream inputStream)
    {
      try (Scanner scanner = new Scanner (inputStream).useDelimiter ("\\A"))
      {
	return scanner.hasNext ()? scanner.next () : "";
      }
    }

    private static void copy (InputStream inputStream,
			      OutputStream outputStream) throws IOException
    {
      byte[] buffer = new byte[1000];
      int length;

      while ((length = inputStream.read (buffer)) > 0)
	{
	  outputStream.write (buffer, 0, length);
	}
    }
  }
}

**Reply by FJBDev** [reply]
Ralph,

Thanks for the info, i will enhance MyTourbook to support crp files zipped with zlib.
Is it the new standard that crp files are now compressed?
Do you think you could provide me 1 or 2 sample files that i can use for the unit tests (note: those files will be public in the MyTourbook github repo)?

Thanks
Frederic

**Reply by WRPSoft** [reply]
Dear Frederic,

Long story short:
There was once the idea to cooperate with an ergometer manufacturer. Binary secure data would have been the prerequisite for this. The project has never been realized and the data was therefore not encrypted. However, some users like the compressed files due to storage space savings, so at some point it was included in the public versions, even to the CicloTour version.
Some time later, text-based files were in fashion again (XML and friends say hello :-) ) and now everything is allowed again (*.fit format). As you do it as a developer, you always get it wrong :)

But please don't ask for details, it's too long ago and there was never the idea of excluding third party programs. At the time, I also sent the relevant information to Hans from the Bikexperience project.

I've attached some *.CRP files for examination. I didn't check if they are all compressed, but I think so. These files are from some backups, I don't have CicloTour installed on my PCs and I cannot take a closer look at the source code. Would otherwise have to reactivate some very old backups, but i would like to avoid that.

Of course you can host these files to the public, it's just simple bike computer data without GPS coordinates. I don't think there are many people who use those file types anymore, but if you can help the few users with relatively little effort, then I don't want to be a spoilsport :-) 

Regards Ralph
[samples.zip](https://github.com/mytourbook/mytourbook/files/9506965/samples.zip)

**Reply by FJBDev** [reply]
Ralph,

Thanks for all this info and details, it helps me understand a bit better the history of the crp format, which is new to me (ive never had a device that generates crp files)
Thanks also for the sample files, it will be very valuable to improve the crp format support in MyTourbook to help the users that use this format and it will be also used for the unit tests as control files

Frederic

**Reply by wolfgang-ch** [reply]
Hans from bikeXperience helped 2007, when I implemented HAC4Pro

https://github.com/mytourbook/mytourbook/blob/b7bbba1f0f1f042f5a47cb1ee23f60611b541808/bundles/net.tourbook.device/src/net/tourbook/device/hac4pro/HAC4ProDeviceDataReader.java#L18-L21

**Comment by ovahead** [comment]
I'm very sorry Frederic!
I said something wrong and forgot the import source of some files. The tour from 080522 and newer were probaply imported from a Ciclotour DAT file and not from CRPs. 
I'm lately a little bit stressed and confused.


Ralph,
thank you very much for your help!

**Reply by FJBDev** [reply]
Ah, that makes sense now, thanks for clarifying :-)

However, thanks to Ralphs files, i will improve the crp format support and will hope to have that done for MyTourbook's next version
@wolfgang-ch do you think its possible to create an issue/feature request from this discussion?

**Reply by wolfgang-ch** [reply]
>@wolfgang-ch do you think its possible to create an issue/feature request from this discussion?

I will do it.

Do you have this option?

![image](https://user-images.githubusercontent.com/1283445/189036881-975729af-6041-4c1f-8b0a-727430d0a76e.png)

**Reply by FJBDev** [reply]
> Do you have this option?
> 

Yes, I see it now and I also see that you created the issue

**Comment by FJBDev** [comment]
@ovahead @WRPSoft 

I have implemented the support of ZLib compressed CRP files and when I tested Ralph's files, I have found something odd that I wanted to ask you about.
When I import Ralph's files a marker is created every 5 seconds which seems odd to me. It seems that it's not a new behavior as I reproduced the issue in 22.8.0.

Is that normal ? Or is it something I should fix for the next version ?

![image](https://user-images.githubusercontent.com/25235665/189534321-1eb2dffa-dd43-494e-ac71-ee2a66b9b02d.png)

**Comment by ovahead** [comment]
Very nice for implenting that!

No that isn't normal. Normally to get a marker in the profile you have to push and hold two buttons at the same on the device. If you only pushed that two buttons the recording ends.
Do you tried my two files 220322 und 080522?

**Reply by FJBDev** [reply]
> Do you tried my two files 220322 und 080522?

I did and I am experiencing the same issue (see my screenshots below), maybe it's a bug in MT and I can investigate a possible fix ? Did you ever have that issue on other `.crp` files ??

![image](https://user-images.githubusercontent.com/25235665/189554018-7d02afe5-a9b9-4bf4-85b3-0b92c35e487d.png)
![image](https://user-images.githubusercontent.com/25235665/189554027-7a9b217f-773b-42ca-92cc-922df62c0914.png)

**Reply by FJBDev** [reply]
@ovahead 

After investigating, it seems that the current code considers a marker if the last column of each row has any text in it.
For the `080522.crp` file (see my screenshots below), the last column contains every time a `0` string and therefore MT considers it is a marker.
Could you give me some files where you have manually created markers so I can see what the actual logic is to detect a real marker ?
Or maybe @WRPSoft  knows also ?

![image](https://user-images.githubusercontent.com/25235665/189554429-a5d6549a-df73-47ee-b753-fc5b1b8fb60a.png)
![image](https://user-images.githubusercontent.com/25235665/189554478-8fb06f6b-2ed6-47cc-90d0-1f85284fc27c.png)

**Reply by WRPSoft** [reply]
Possibly due to the fact that after the comment/lap column, a column for cadence has been added at some time. If you test during parsing whether the lap column contains a string, then the new CAD column may be interpreted as a comment/lap. But I don't your code, just a possibilty.

**Reply by wolfgang-ch** [reply]
I don't have .crp files with 0 at the end, as you can see in the above example https://user-images.githubusercontent.com/1283445/188548572-4513eda2-4a74-4597-bb00-08eaa7c35955.png, therefor all data after the timestamp was interpreted as a marker

**Reply by wolfgang-ch** [reply]
I found 2 examples with a marker

![image](https://user-images.githubusercontent.com/1283445/189587613-e6089fde-c475-4ddd-87e7-dead0eef097b.png)

![image](https://user-images.githubusercontent.com/1283445/189587631-c02ef0ef-e12a-4256-8bd1-333bf9ca9dc5.png)

**Reply by WRPSoft** [reply]
As written before, beginning with file version 10 (first number in the second line of the crp file), a new column for cadence values has been added.

case vers of
                  // CicloTour Vers. 2.0
                10, 11:
                  begin
                    ParseData(9, ''#9'', strData, aValues);
                    HRMData.CAD[idx] := StrToInt(aValues[10]);
                  end;

**Reply by wolfgang-ch** [reply]
>first number in the second line of the crp file

My files are from version 9

![image](https://user-images.githubusercontent.com/1283445/189609154-2a991abd-14b3-4254-bfa1-5c7848368180.png)

![image](https://user-images.githubusercontent.com/1283445/189609189-edbc3377-9f7f-45d7-99bb-0fbe435fb702.png)

**Reply by WRPSoft** [reply]
Then everything should be clarified now and Frederic should be able to easily make the necessary adjustments :-)

**Reply by FJBDev** [reply]
Thanks, I have now added support for files of which the version is > 9 and it looks good now

Thanks @WRPSoft  for the details on the new cadence column
Can you do me one more favor and give me 1-2 `.crp` files of version <= 9 for the unit tests to ensure that the import of those versions is unaltered ? Thanks!

![image](https://user-images.githubusercontent.com/25235665/189728842-8b2e40c8-dc54-4f2f-82a3-7cef4169e524.png)

**Comment by ovahead** [comment]
Frederic. 
I forgot to mention that always every file has at least one marker. it's the start marker. Look at the picture.
Ive done one extra marker manually with the software CicloTour (xxxxx)   
![170622](https://user-images.githubusercontent.com/112829463/189589979-a1fb8dd5-72a0-4186-b3c0-ca705a8a9a13.jpg)
[170622_extra_marker_manuell.txt](https://github.com/mytourbook/mytourbook/files/9545365/170622_extra_marker_manuell.txt)

Changed CRP to TXT...

**Reply by FJBDev** [reply]
Thank you, I have added the creation of this first marker as it is not created in the current MT version.

Thanks for the `170622_extra_marker_manuell.txt` file example.

>Changed CRP to TXT...
By the way, changing the extension to txt doesn't help as the content of the file is unchanged. It is better to keep the original `.crp` extension

![image](https://user-images.githubusercontent.com/25235665/189729449-0a6a5a9a-253d-4f73-8545-53f26f86384f.png)

**Reply by FJBDev** [reply]
@ovahead Do you have `.crp` files that contain cadence values?

**Reply by WRPSoft** [reply]
This is the only file with Cadence values that I could quickly find. When I added the Cadence to this format, I was primarily using other bike computers.

By the way: if you can remember the old Hac data format, you may still know that the cadence, analogous to the temperature, was originally only saved in 2-minute intervals and then had to be graded more finely by interpolation, which was the reason that I've never used this parameter because I never saw any real usage for it. Cadence - in contrast to temperature - is normally a very volatile value.

But now I have to take care of my own projects a little more. You should now know all you must know about the crp format.

PS: Sorry, I don't have any old crp data anymore, they were all converted to the current 10 or 11 file format at some point. But Wolfgang has some CRP FileVersion 9 data, so that shouldn't be a problem.
[080510.zip](https://github.com/mytourbook/mytourbook/files/9553932/080510.zip)

**Reply by FJBDev** [reply]
@WRPSoft 
Thank you for this file, I was able to validate that the cadence values correctly import.

![image](https://user-images.githubusercontent.com/25235665/189943594-415a83ee-eaba-48e9-8d32-90552554f05e.png)

>PS: Sorry, I don't have any old crp data anymore, they were all converted to the current 10 or 11 file format at some point. But Wolfgang >has some CRP FileVersion 9 data, so that shouldn't be a problem.

@wolfgang-ch Do you think you have a v9 file I could use in the unit tests ?

**Reply by wolfgang-ch** [reply]
Here are two files from the beginning of my mountain bike career :-)
[2004-07-03.crp.txt](https://github.com/mytourbook/mytourbook/files/9563112/2004-07-03.crp.txt)
[2004-07-25.crp.txt](https://github.com/mytourbook/mytourbook/files/9563110/2004-07-25.crp.txt)

**Reply by FJBDev** [reply]
Thank you! I was able to ensure that the improved CRP import hasn't altered the import of files <= v9 👍

**Comment by ovahead** [comment]
You'r welcome!

I changed the extensions only for uploading reason and to avoid to zip it, because crp isn't supported by github.
I'm sorry a have not files with cadence values. Never used a cadence meter. Maybe WERPSoft?

**Reply by FJBDev** [reply]
> I changed the extensions only for uploading reason and to avoid to zip it, because crp isn't supported by github

Ah ok, I was not aware of that, thanks!

**Comment by FJBDev** [comment]
@ovahead 
As I am pretty much done implementing all those improvements/fixes (at this point, I am only missing a v9 file to ensure I didn't break the previous imports), would you be interested in having access to a beta version of MT to ensure that all your `.crp` files import correctly ?

**Comment by ovahead** [comment]
Sure, I'm interested.

edited:
BTW
What is the meaning of a v9 file? Is Ciclotour meant?

**Reply by FJBDev** [reply]
> edited: BTW What is the meaning of a v9 file? Is Ciclotour meant?

See our discussion above and screenshots. A V9 file is apparently an older `.crp` file.

**Reply by FJBDev** [reply]
> Sure, I'm interested.
> 
You can download a test version [here](https://www.dropbox.com/s/34wrdlyixm5f41g/mytourbook-22.8.0-win-setup-CRPImprovements.exe?dl=0)
Let me know if that resolves your import issues!

**Reply by ovahead** [reply]
Sorry, I think I need a version with java again. This version isn't starting on my system.

**Reply by FJBDev** [reply]
> Sorry, I think I need a version with java again. This version isn't starting on my system.

Sure thing ! [here](https://www.dropbox.com/s/m9vlthc3i89fphj/mytourbook-22.8.0-win-setup-CRPImprovements-with-java-runtime.exe?dl=0) is a downloadable version with java embedded

**Reply by ovahead** [reply]
First look are great.

But the markers has no distance or wrong values. They weren't there before... 
Pictures are from a newer version of the CRPs (zlib).
Older CRPs (ANSI) has now at least on marker (the start marker) or just one marker more. Also without or wrong distance values.

![Marker](https://user-images.githubusercontent.com/112829463/190280910-ad51931c-7f82-4b7d-8211-231d014fe4f4.jpg)
![Marker2](https://user-images.githubusercontent.com/112829463/190280920-2d3f3e7d-dcf1-4982-bd4d-6a5deb0f0491.jpg)

**Reply by FJBDev** [reply]
I will investigate the wrong distance values. This is not something I modified so it's possible it's an existing bug. Do you see this issue in MT <= 22.8.0 as well ?

**Reply by ovahead** [reply]
I'm not really sure. I think it was as well

**Reply by FJBDev** [reply]
@ovahead 
After investigation, it was what I was thinking, an existing bug.
I have fixed it, see my screenshots below.

I will generate a new test version and will give you the link when it's ready so you can test again.
Thanks for finding this issue, it's always better to find bugs before the official release 👍 

I am surprised that nobody found this issue after all those years, maybe not a lot of people use the `.crp` format ?
@wolfgang-ch You used the `.crp` format in the past, did you notice this bug with your `.crp` files ? If not, you might want to reimport them to fix this marker distance issue ?

**Before the fix:**
![image](https://user-images.githubusercontent.com/25235665/190449346-b7bb1b9c-7ac3-46e4-ac19-0271eb338697.png)

**After the fix:**
![image](https://user-images.githubusercontent.com/25235665/190449526-d6988b22-0cd6-4ed6-9d98-b1d052c0ca7c.png)

**Reply by ovahead** [reply]
Looks pretty good!

**Reply by FJBDev** [reply]
@ovahead 
[Here](https://www.dropbox.com/s/ciitl1iae45tpp6/mytourbook-22.CRPFIXES-win-setup-with-java-runtime.exe?dl=0) is a new beta version you can download that contains the fix for the tour marker distance

**Reply by ovahead** [reply]
Looks perfect, on both CRP old and new.
Thank you very very much for all your work to make it possible to import all CRPs!

Maybe only the distance for the start markers are missing. But its questionable if there must be a value of 0,00 because it's obvious...

**Reply by FJBDev** [reply]
> Looks perfect, on both CRP old and new. Thank you very very much for all your work to make it possible to import all CRPs!

Thanks for testing, I am glad it's working good now.
I have created a PR so that all my CRP improvements can be in the next version of MyTourbook

> Maybe only the distance for the start markers are missing. But its questionable if there must be a value of 0,00 because it's obvious...

Well, the start markers are at the start so their distance is 0 and I think that's why MT displays blank in the Tour Marker view column.
I have tested on a tour and created a marker at the very beginning (1st time slice) and I get the same behavior, the distance for the first marker is empty because it's equal to 0

**Reply by ovahead** [reply]
It really doesn't matter!
And thanks again for the (hard?) work!

**Reply by FJBDev** [reply]
> It really doesn't matter! And thanks again for the (hard?) work!

You're very welcome

**Reply by wolfgang-ch** [reply]
>@wolfgang-ch You used the `.crp` format in the past, did you notice this bug with your `.crp` files ? If not, you might want to reimport them to fix this marker distance issue ?

I used only a few markers but after 18 years (2004) I cannot remember why I created the markers

**Reply by ovahead** [reply]
Frederic,

it seems there is still a little issue with CRP Files, In the rare case the pause time is greater as the movement time (especially if you forgot to end the excersize then the device stopped only after 24 hours or so automatic. With potential hours of "pause time") The times were not shown right in MTB respectively there were not calculated correctly. Please have a look at the pictures.

If I delete the "pause time" at the end point by point (sometimes more than 3500 points or so) with CicloTour, the data from an imported CRP looks okay.

![Times](https://user-images.githubusercontent.com/112829463/192108965-39d5cbab-3242-4434-8a46-1a17a80f4054.jpg)
[280412crp.txt](https://github.com/mytourbook/mytourbook/files/9639149/280412crp.txt)

Same Tour in CicloTour:
![CicloTour-Tourdata](https://user-images.githubusercontent.com/112829463/192108992-3de7981f-7acd-4c65-b367-c6b22c2ca92b.jpg)

**Reply by FJBDev** [reply]
@ovahead 

Thanks for finding this issue and providing the file.
I believe I have found the issue. Looking at your provided file, it seems that your activity overlapped 2 days and the below lines explain why you are seeing an erroneous elapsed time and moving time etc...
This is a bug I will investigate and attempt to fix

```
0	0	3067	93	0	0	19,0	23:59:40		0
0	0	3067	94	0	0	19,0	00:00:00		0
```

**Reply by FJBDev** [reply]
@ovahead 

I found a fix for the issue above and this is what the times (elapsed, recorded, moving....) look like with the fix.
What do you think ?

![image](https://user-images.githubusercontent.com/25235665/192121918-26931414-f064-45c2-96e5-6894accb862a.png)

**Reply by ovahead** [reply]
According to Ciclotour the moving time is only 1:35:00 instead of 1:55:20. See also snapshot from CicloTour. Because of this should the break time 22:24:55?

I would changed the crps but its a lot of work to change several files an then only point by point :(

**Reply by FJBDev** [reply]
> According to Ciclotour the moving time is only 1:35:00 instead of 1:55:20. See also snapshot from CicloTour. Because of this should the break time 22:24:55?

Not necessarily, because my hunch is that Ciclotour moving time algorithm is different than MT's [computed time](https://mytourbook.sourceforge.io/mytourbook/index.php/documentation/preferences/general/pref-computed-values#break-time)
You would need to try playing with the computed time settings in MT I believe

**Reply by ovahead** [reply]
Thanks, for the hint and link.

Hm, I think this overhelms me a bit ;)
Anyway I tried some settings but after reimporting the times are all the same... Whatever method I choose, only speed or speed and  ⌀-Speed and which values I choose.  Is there anything that I left that MT not recomputing the values (only of the current choosed tour)?
And isn't it not only for GPS devices?

**Reply by FJBDev** [reply]
>  Anyway I tried some settings but after reimporting the times are all the same... Whatever method I choose, only speed or speed and ⌀-Speed and which values I choose. Is there anything that I left that MT not recomputing the values (only of the current choosed tour)? And isn't it not only for GPS devices?

I tried as well the different break methods and indeed, the moving time is still 1:55:20.
Not sure how to explain the difference with CicloTour since I don't know how CicloTour computed its own moving time

**Reply by FJBDev** [reply]
Actually, I just remember now that it's normal that the break time is unchanged because your file doesn't have GPS.

As you can see in the MT UI, the break time recomputation is for tours created by a GPS device

![image](https://user-images.githubusercontent.com/25235665/193432526-b1745f2d-98e0-45df-aa5b-30da048de19f.png)
